### PR TITLE
Support parametrized VARCHAR in blackhole connector

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

**Comment by losipiuk** [comment]
@arhimondr, @kokosing take a look

File: presto-spi/src/main/java/com/facebook/presto/spi/type/VarcharType.java [hunk 1]
```diff
@@ -29,6 +29,11 @@
     public static final int MAX_LENGTH = Integer.MAX_VALUE;
     public static final VarcharType VARCHAR = new VarcharType(MAX_LENGTH);
 
+    public static VarcharType createUnboundedVarcharType()
```
Comment (fiedukow) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/VarcharType.java]: This seems to be invalid name, this method does not create anything.
Why not just just VARCHAR constant btw?

**Comment by fiedukow** [comment]
LGTM % comment

File: presto-spi/src/main/java/com/facebook/presto/spi/type/VarcharType.java (hunk 1 above)
Comment (losipiuk) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/VarcharType.java]: This method was actually introduced by @arhimondr in another commit (not merged yet).
The fact that it does not create anything is not important - it is just implementation detail  - and I assumed that we want to get rid of the variable eventually, replacing it with more explicitly named type constructor.

Comment (fiedukow) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/VarcharType.java]: It's implementation detail only as long as the object is immutable - which is the case right now.
I will introduce this function into VARCHAR(x) functions as well than ;)

File: presto-blackhole/src/main/java/com/facebook/presto/plugin/blackhole/BlackHolePageSourceProvider.java [hunk 2]
```diff
@@ -73,9 +70,15 @@ public ConnectorPageSource createPageSource(
                 blackHoleSplit.getPagesCount()));
     }
 
-    private Page generateZeroPage(List<Type> types, int rowsCount, int fieldsLength)
+    private boolean isSupportedType(Type type)
```
Comment (kokosing) [inline presto-blackhole/src/main/java/com/facebook/presto/plugin/blackhole/BlackHolePageSourceProvider.java]: put it under the usage

**Comment by kokosing** [comment]
lgtm

**Comment by losipiuk** [comment]
@Martint, no brainer, please merge when you have a chance.

**Comment by martint** [comment]
Can you check the build failures?

**Comment by martint** [comment]
The failures looked unrelated. I kicked off another travis build.

It that passes, I'll merge it after we're done with the current release.

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**